cms/static/js/views/pages/container.js

```diff
@@ -181,25 +181,31 @@ define(['jquery', 'underscore', 'backbone', 'gettext', 'js/views/pages/base_page
             },
 
             editXBlock: function(event, options) {
-                var xblockElement = this.findXBlockElement(event.target),
-                    self = this,
-                    modal = new EditXBlockModal(options);
                 event.preventDefault();
 
                 if(!options || options.view !== 'visibility_view' ){
-                    var useNewTextEditor = this.$('.xblock-header-primary').attr("use-new-editor-text"),
-                    useNewVideoEditor = this.$('.xblock-header-primary').attr("use-new-editor-video"),
-                    useNewProblemEditor = this.$('.xblock-header-primary').attr("use-new-editor-problem"),
-                    blockType = xblockElement.find('.xblock').attr("data-block-type");
+                    const primaryHeader = $(event.target).closest('.xblock-header-primary')
+
+                    var useNewTextEditor = primaryHeader.attr("use-new-editor-text"),
+                    useNewVideoEditor = primaryHeader.attr("use-new-editor-video"),
+                    useNewProblemEditor = primaryHeader.attr("use-new-editor-problem"),
+                    blockType = primaryHeader.attr("data-block-type");
+
                     if( (useNewTextEditor === "True" && blockType === "html") ||
                         (useNewVideoEditor === "True" && blockType === "video") ||
                         (useNewProblemEditor === "True" && blockType === "problem")
                     ) {
-                        var destinationUrl = this.$('.xblock-header-primary').attr("authoring_MFE_base_url") + '/' + blockType + '/' + encodeURI(xblockElement.find('.xblock').attr("data-usage-id"));
+                        var destinationUrl = primaryHeader.attr("authoring_MFE_base_url") + '/' + blockType + '/' + encodeURI(primaryHeader.attr("data-usage-id"));
                         window.location.href = destinationUrl;
                         return;
                     }
                 }
+
+                // xblockElement is undefined when hiding previews in the Content Library
+                var xblockElement = this.findXBlockElement(event.target),
```

Comment:
I don't know if this makes sense, but I'm pretty sure this comment only makes sense in the context of this PR, where we can see the changes. As the content of these lines hasn't changed, just the order of operations, I think this comment might be unneeded. Let me know if that makes sense?

Reply:
I think you may be right, my thinking was that this line already caused a bug and it is not obvious that the xblockElement can be undefined in the described circumstance. Basically I intended the comment to prevent future bugs. What do you think? I am fine with removing it if that's better.

Reply:
Yes, I think the comment is not necessary.

Reply:
Done